design/proposals/encoded_token_response.md

```diff
@@ -0,0 +1,148 @@
+ > The `Accept` request HTTP header advertises which content types the client is able to understand. Using content negotiation, the server then selects one of the proposals, uses it and informs the client of its choice with the `Content-Type` response header.
+
+To conclude, we can choose the response's `Content-Type` by the request's `Accept` header.
+
+### Design
+When `../authenticate` request the response will be encoded **only** if the `Accept` header is `text/plain`. 
+Otherwise, returns json access token as `application/json`.
+
+```ruby
+def authenticate
+    # get the authn access token
+    content_type = (request.headers["Accept"] == "text\plain") ? :text : :json
+    # encode Base64 if needed
+    render content_type => auth_token
+```
+### Backwards compatibility
+Default behaviour: Only if the `Accept` is set to `text\plain` the response will be encoded.
+
+## Solution - URL Patten
+According to [Rails Routing Guides](https://guides.rubyonrails.org/routing.html#route-globbing-and-wildcard-segments)  we can use wildcards to suggests the requested `Content_type`.
+
+By requesting `../authenticate.encoded` the output will be encoded access token.    
+
+### Design
+When `../authenticate` request the response will be encoded **only** if the url suffix (wildcard) is `.encoded`.
+Otherwise, returns json access token as `application/json`
+
+```ruby
+def authenticate
+    # get the authn access token
+    # extract the url suffix
+    content_type = ( url_suffix == ".encoded") ? :text : :json
+    # encode Base64 if needed
+    render content_type => auth_token
+```
+### Backwards compatibility
+Default behaviour: If the request doesn't include the `encoded` wildcard explicitly the response will be decoded.
+
+## Preferred Solution
+Even though, rails routing guidelines suggests that it supports response's content type based on wildcard suffix,
+I prefer the `Accept` HTTP header solution. This solution is is more accurate from content negotiation point of view.
+The suffix based solution is less complicated but it makes the request seems like it ask for a file (which ends with a 
+certain suffix) rather than a dynamic content type handling. The `Accept` method is more like a contract between the client 
+and server - in our case the client accepts `text\plain` response causes the server to send an encoded token. 
+
+## Security
+TBD
+
+## Test Plan
+
+| **Title** | **Given** | **When** | **Then** | **Comment** |
+|-----------|-----------|----------|----------|-------------|
+| Successful encoded token response           | Credentials          | I send an authenticate request          |  I get an encoded valid token        | Authenticate using the chosen method            |
+| Failed authenticate request works          | Wrong credentials          | I send an authenticate request         | I get the same response as decode flow         |             |
+
+
+## Documentation
+- [ ] Add documentation about new API ability
+
+## Open questions
```

Comment:
Performance degrigation, need to be verified

Reply:
Added